components/popover_list_members/popover_list_members.tsx

```diff
@@ -0,0 +1,293 @@
+            onItemClick={handleShowDirectChannel}
+            showMessageIcon={props.currentUserId !== user.id && !isDirectChannel}
+            status={props.statuses[user.id]}
+            user={user}
+        />
+    ));
+
+    const channelIsArchived = props.channel.delete_at !== 0;
+    let popoverButton;
+    let handleButtonOnClick = showMembersModal;
+    let editButton = null;
+    if (props.channel.type !== Constants.GM_CHANNEL && !channelIsArchived) {
+        let membersName = (
+            <FormattedMessage
+                id='members_popover.manageMembers'
+                defaultMessage='Manage Members'
+            />
+        );
+
+        const isDefaultChannel = props.channel.name === Constants.DEFAULT_CHANNEL;
+
+        if (isDefaultChannel || !props.manageMembers) {
+            membersName = (
+                <FormattedMessage
+                    id='members_popover.viewMembers'
+                    defaultMessage='View Members'
+                />
+            );
+        }
+
+        if (props.addMembersABTest === AddMembersToChanneltreatments.BOTTOM && props.manageMembers) {
+            handleButtonOnClick = () => onAddNewMembersButton(AddMembersToChanneltreatments.BOTTOM);
+            membersName = (
+                <FormattedMessage
+                    id='members_popover.addMembers'
+                    defaultMessage='Add Members'
+                />
+            );
+            editButton = (
+                <button
+                    className='btn btn-link'
+                    id='editBtn'
+                    onClick={showMembersModal}
+                >
+                    <i className='icon icon-pencil-outline'/>
+                    <FormattedMessage
+                        id='members_popover.editBtn'
+                        defaultMessage='Edit'
+                    />
+                </button>
+            );
+        } else if (props.addMembersABTest === AddMembersToChanneltreatments.TOP && props.manageMembers) {
+            editButton = (
+                <button
+                    className='btn btn-link'
+                    id='addBtn'
+                    onClick={() => onAddNewMembersButton(AddMembersToChanneltreatments.TOP)}
+                >
+                    <i className='icon icon-account-plus-outline'/>
+                    <FormattedMessage
+                        id='members_popover.add'
+                        defaultMessage='Add'
+                    />
+                </button>
+            );
+        }
+
+        popoverButton = (
+            <div
+                className='more-modal__button'
+                key={'popover-member-more'}
+            >
+                <button
+                    className='btn btn-link'
+                    data-testid='membersModal'
+                    onClick={handleButtonOnClick}
+                >
+                    {membersName}
+                </button>
+            </div>
+        );
+    }
```

Comment:
this whole section seems a bit too complicated ... could you try and find a way to simplify it? If you need help I can have a look as well, just hit me a DM! :D

Reply:
I will look into this and come back to you with a suggestion

Reply:
if you need help feel free to send me a DM